app/main.py

```diff
@@ -0,0 +1,178 @@
+import json
+import os
+
+from typing import List, Optional
+
+import pandas as pd
+import numpy as np
+import uvicorn
+from SBTi.interfaces import PortfolioCompany, ScenarioInterface
+
+from fastapi import FastAPI, File, Form, UploadFile, Body, HTTPException
+from pydantic import BaseModel
+import mimetypes
+import SBTi
+from SBTi.portfolio_aggregation import PortfolioAggregationMethod
+
+app = FastAPI(
+    title="SBTi Finance Temperature Alignment tool API",
+    description="This tool helps companies and financial institutions to assess the temperature alignment of current "
+                "targets, commitments, and investment and lending portfolios, and to use this information to develop "
+                "targets for official validation by the SBTi.",
+    version="0.1.0",
+)
+
+
+mimetypes.init()
+UPLOAD_FOLDER = 'data'
+
+with open('config.json') as f_config:
+    config = json.load(f_config)
+
+
+class RequestTemperatureScore(BaseModel):
+    data_providers: Optional[List[str]] = []
+    companies: List[PortfolioCompany]
+    default_score: float
+    aggregation_method: Optional[str] = "WATS"
+    grouping_columns: Optional[List[str]] = None
+    include_columns: Optional[List[str]] = []
+    scenario: Optional[ScenarioInterface] = None
+    anonymize_data_dump: Optional[bool] = False
+    filter_scope_category: Optional[List[str]] = []
+    filter_time_frame: Optional[List[str]] = []
+
+
+class ResponseTemperatureScore(BaseModel):
+    aggregated_scores: dict
+    scores: List[dict]
+    coverage: float
+    companies: List[dict]
+    feature_distribution: Optional[dict]
+
+
+@app.post("/temperature_score/", response_model=ResponseTemperatureScore)
+def calculate_temperature_score(
+        companies: List[PortfolioCompany] = Body(..., description="A portfolio containing the companies"),
+        default_score: float = Body(
+            default=config["default_score"],
+            gte=0,
+            description="The default score to fall back on when there's no target available."),
+        data_providers: Optional[List[str]] = Body(
+            default=[],
+            description="A list of data provider names to use. These names should be available in the list that can be "
+                        "retrieved through the /data_providers/ endpoint."),
+        aggregation_method: Optional[str] = Body(
```

Comment:
why not use an enum for this field? then it will also validate it automatically that it actually is a valid choice. will also make the code nicer where you have the if statements